core/state.go

```diff
@@ -400,12 +400,16 @@ func (s *State) updateContractStorages(stateTrie *trie.Trie, diffs map[felt.Felt
 	})
 
 	// update per-contract storage Tries concurrently
-	contractUpdaters := pool.NewWithResults[*db.BufferedTransaction]().WithErrors().WithMaxGoroutines(runtime.GOMAXPROCS(0))
+	contractUpdaters := pool.NewWithResults[*db.BufferedTransactionWithAddress]().WithErrors().WithMaxGoroutines(runtime.GOMAXPROCS(0))
 	for _, key := range keys {
-		conractAddr := key
-		updateDiff := diffs[conractAddr]
-		contractUpdaters.Go(func() (*db.BufferedTransaction, error) {
-			return s.updateStorageBuffered(&conractAddr, updateDiff, blockNumber, logChanges)
+		contractAddr := key
+		updateDiff := diffs[contractAddr]
+		contractUpdaters.Go(func() (*db.BufferedTransactionWithAddress, error) {
+			result, err := s.updateStorageBuffered(&contractAddr, updateDiff, blockNumber, logChanges)
```

Comment:
You already have access to the address when calling `s.updateStorageBuffered`, there is no need to return the address inside the result again.